src/Product/addImage.ts

```diff
@@ -0,0 +1,46 @@
+import assert from 'assert';
+import { HttpModule } from '@whppt/api-express';
+import { Product, WhpptProductImageData } from './Models/Product';
+
+export type ChangeProductImageArgs = {
+  domainId: string;
+  productId: string;
+  featureImageId?: string;
+  image: WhpptProductImageData;
+};
+
+const changeImage: HttpModule<ChangeProductImageArgs, void> = {
+  authorise({ $identity }, { user }) {
+    return $identity.isUser(user);
+  },
+  exec({ $database, createEvent }, { domainId, productId, image, featureImageId }) {
+    assert(domainId, 'Product requires a Domain Id.');
```

Comment:
Should really read Adding an image to a product requires a domain id.

But I think we could just go generic: "DomainId is required" ?